test/email_validator_test.rb

```diff
@@ -3,96 +3,97 @@
 
 class EmailValidatorTest < Minitest::Test
 
-  def user_class(
-    smtp: true, states: %i[deliverable risky unknown], free: true, role: true,
-    accept_all: true, disposable: true, timeout: 3, **options
-  )
-    Class.new do
-      include ActiveModel::Model
-      attr_accessor :email, :email_verification_result
-
-      validates :email, presence: true, email: {
-        smtp: smtp, states: states,
-        free: free, role: role, disposable: disposable, accept_all: accept_all,
-        timeout: timeout
-      }.merge(options)
-
-      def self.name
-        'TestClass'
-      end
-
-      # stub changes to always be true
-      def changes
-        { email: true }
-      end
-    end
-  end
-
   def setup
     Emailable.api_key = 'test_7aff7fc0142c65f86a00'
-    sleep(0.25)
   end
 
   def test_valid
-    @user = user_class.new(email: 'deliverable@example.com')
+    valid_user = user_class.new(email: 'deliverable@example.com')
 
-    assert @user.valid?
-    assert @user.errors.empty?
+    assert valid_user.valid?
+    assert valid_user.errors.empty?
   end
 
   def test_invalid
-    @user = user_class.new(email: 'undeliverable@example.com')
+    invalid_user = user_class.new(email: 'undeliverable@example.com')
 
-    assert !@user.valid?
-    assert @user.errors[:email].present?
+    refute invalid_user.valid?
+    assert invalid_user.errors[:email].present?
   end
 
   def test_verification_result
-    @user = user_class.new(email: 'undeliverable@example.com')
-    @user.valid?
+    invalid_user = user_class.new(email: 'undeliverable@example.com')
+    invalid_user.valid?
 
-    refute_nil @user.email_verification_result
-    assert @user.email_verification_result.state, :undeliverable
+    refute_nil invalid_user.email_verification_result
+    assert_equal 'undeliverable', invalid_user.email_verification_result.state
   end
 
   def test_boolean_options
     %i[smtp free role disposable accept_all].each do |option|
-      invalid_user = user_class(option => 'string').new
-      valid_user = user_class.new
+      invalid_options = user_class(option => 'string').new
+      valid_options = user_class.new
 
-      assert !valid_user.valid?
-      assert_raises(ArgumentError) { invalid_user.valid? }
+      refute valid_options.valid?
```

Comment:
Why are `valid_options` not valid?

Reply:
Because `valid?` runs the validation and doesn't have anything to do with the options for the validation.